dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/NettyFrameworkImpl.java

```diff
@@ -101,40 +105,191 @@ public class NettyFrameworkImpl implements ServerQuiesceListener, NettyFramework
+     */
+    private AutoScalingEventExecutorChooserFactory createThreadScaler(Map<String, Object> config) {
+        if(config == null) {
+            throw new IllegalArgumentException("Passed config object that was null!!");
+        }
+        int minThreads, maxThreads, upStep, downStep, cycles;
+        long windowSize;
+        double downThreshold, upThreshold;
+        minThreads = (Integer)config.getOrDefault(NettyConstants.SCALER_MIN_THREADS_PROPERTY, NettyConstants.SCALER_MIN_THREADS);
+        maxThreads = (Integer)config.getOrDefault(NettyConstants.SCALER_MAX_THREADS_PROPERTY, NettyConstants.SCALER_MAX_THREADS);
+        upStep = (Integer)config.getOrDefault(NettyConstants.SCALER_UP_STEP_PROPERTY, NettyConstants.SCALER_UP_STEP);
+        downStep = (Integer)config.getOrDefault(NettyConstants.SCALER_DOWN_STEP_PROPERTY, NettyConstants.SCALER_DOWN_STEP);
+        cycles = (Integer)config.getOrDefault(NettyConstants.SCALER_CYCLES_PROPERTY, NettyConstants.SCALER_CYCLES);
+        windowSize = (Long)config.getOrDefault(NettyConstants.SCALER_WINDOW_PROPERTY, NettyConstants.SCALER_WINDOW);
+        downThreshold = (Double)config.getOrDefault(NettyConstants.SCALER_DOWN_THRESHOLD_PROPERTY, NettyConstants.SCALER_DOWN_THRESHOLD);
+        upThreshold = (Double)config.getOrDefault(NettyConstants.SCALER_UP_THRESHOLD_PROPERTY, NettyConstants.SCALER_UP_THRESHOLD);
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "Creating AutoScaler from config with minThreads: " + minThreads + ", maxThreads: " + maxThreads + ", windowSize: " + windowSize + ", downThreshold: " + downThreshold + ", upThreshold: " + upThreshold + ", upStep: " + upStep + ", downStep: " + downStep + ", cycles: " + cycles);
+        }
+        return new AutoScalingEventExecutorChooserFactory(minThreads, maxThreads, windowSize, TimeUnit.MILLISECONDS, downThreshold, upThreshold, upStep, downStep, cycles);
+    }
+
+    /**
+     * Method for setting Netty system properties default if not already set.
+     */
+    private void setNettySystemProperties() {
+        // If the properties are not already manually set in the system, set the defaults
+        // tested to bring the best performance
+        if (System.getProperty("io.netty.leakDetection.level") == null) {
+            // Netty's buffer leak detection is enabled by default which is unnecessary unless
+            // issues are found and leak detection is needed to debug them.
+            System.setProperty("io.netty.leakDetection.level", "DISABLED");
+        }
+        if (System.getProperty("io.netty.allocator.type") == null) {
+            // On Netty version 4.2, the default allocator changed from pooled to adaptive and so
+            // as of that moment, the adaptive allocator showed a regression from the pooled allocator
+            // used in Netty 4.1. We switch to the pooled allocator for the moment until these issues
+            // are addressed
```

Comment:
@isaacrivriv, can you open an issue for this so we can revisit it once the bug is solved? I don't want to forget!